src/PufferVaultMainnet.sol

```diff
@@ -80,20 +105,107 @@ contract PufferVaultMainnet is PufferVault {
         }
 
         uint256 assets = previewRedeem(shares);
+
+        _checkDailyWithdrawalLimits(assets); //@todo figure if it is better to override `maxWithdraw`
+
+        _wrapETH(assets);
+
         // solhint-disable-next-line func-named-parameters
         _withdraw(_msgSender(), receiver, owner, assets, shares);
 
         return assets;
     }
 
-    // slither-disable-next-line dead-code
-    function _wrap(uint256 amount) internal {
-        _WETH.deposit{ value: amount }();
+    /**
+     * @param withdrawalAmount is the assets amount, not shares
+     */
+    function _checkDailyWithdrawalLimits(uint256 withdrawalAmount) internal {
+        VaultStorage storage $ = _getPufferVaultStorage();
+
+        // If daily withdrawal limit is 0, then there is no limit
+        if ($.dailyWithdrawalLimit == 0) {
```

Comment:
dailyWithdrawalLimit should be maxUInt instead for no limit case. (dailyWithdrawalLimit == maxUint --> return ) 

(maybe set `dailyWithdrawalLimit` in initializer to 0 for explicit declaration)

Reply:
I think this is also fixed by changing the logic to use default 4626 limits, I'll resolve this after reviewing the new flow.